src/dsp_tools/commands/xml_validate/reformat_input.py

```diff
@@ -0,0 +1,194 @@
+from typing import Callable
+from typing import Sequence
+
+from lxml import etree
+from rdflib import XSD
+from rdflib import Literal
+from rdflib import URIRef
+
+from dsp_tools.commands.xml_validate.models.data_rdf import BooleanValueRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import ColorValueRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import DataRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import DateValueRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import DecimalValueRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import GeonameValueRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import IntValueRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import LinkValueRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import ListValueRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import ResourceRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import RichtextRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import SimpleTextRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import TimeValueRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import UriValueRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import ValueRDF
+
+
+def transform_into_data_deserialised(root: etree._Element) -> DataRDF:
+    """
+    Takes the root of an XML
+    Extracts the metadata of the project and transforms all its resources.
+
+    Args:
+        root: root of an xml with qnames and comments removed
+
+    Returns:
+        Class instance with the information reformatted
+    """
+    shortcode = root.attrib["shortcode"]
+    default_ontology = root.attrib["default-ontology"]
+    resources = _deserialise_all_resources(root)
+    return DataRDF(shortcode=shortcode, default_onto=default_ontology, resources=resources)
+
+
+def _deserialise_all_resources(root: etree._Element) -> list[ResourceRDF]:
+    all_res = []
+    for res in root.iterchildren():
+        match res.tag:
+            case "resource":
+                all_res.append(_deserialise_one_resource(res))
+            case "annotation" | "region" | "link" | "video-segment" | "audio-segment":
+                all_res.append(_deserialise_dsp_base_resource(res))
+            case _:
+                pass
+    return all_res
+
+
+def _deserialise_dsp_base_resource(resource: etree._Element) -> ResourceRDF:
+    return ResourceRDF(
+        res_id=URIRef(resource.attrib["id"]),
+        res_class=URIRef(resource.attrib["restype"]),
```

Comment:
The special tags don't have a "restype" attribute. Have you tried this out? I think it wouldn't work

Reply:
They do. In the xmlupload preparation step, we add the restype to the elements that are for example `<region>`. This is something that should be changed in the future, in the xmlupload too. When we implement the dsp-base resources in SHACL we will have a separate class for each. So this will be deleted, which is why I am ok having this sub-par handling here. Until then, this is the easiest way, because we need them as targets

Reply:
But wait, I think this entire function is wrong and will actually never be called. Because in the preparation step you mentioned, not only the restypes are added, but the special tag names are transformed to `<resource>`. See https://github.com/dasch-swiss/dsp-tools/blob/45af782b15c9d0703daf6ee0e76066331632b95c/src/dsp_tools/utils/xml_utils.py#L35

That means:
 - This line is never reached: https://github.com/dasch-swiss/dsp-tools/blob/45af782b15c9d0703daf6ee0e76066331632b95c/src/dsp_tools/commands/xml_validate/reformat_input.py#L49
 - the function `_deserialise_dsp_base_resource()` is never called, and can be deleted.

(Anyway, `_deserialise_dsp_base_resource()` ignores the values, which is not okay. But this doesn't matter any more).

Or do I miss something?